hibernate-core/src/main/java/org/hibernate/boot/spi/BasicTypeRegistration.java

```diff
@@ -25,7 +29,23 @@ public BasicTypeRegistration(BasicType basicType) {
 
 	public BasicTypeRegistration(BasicType basicType, String[] registrationKeys) {
 		this.basicType = basicType;
-		this.registrationKeys = registrationKeys;
+		Class returnedClass = basicType.getReturnedClass();
+
+		if ( returnedClass != null ) {
```

Comment:
Is and should this handling be dependent of `@TypeDef(defaultForType = Clasz.class)` ? And specifically do not register if its not set as a default for any type? To prevent possible collisions between types between custom basic types?

Reply:
I think it should work for TypeContributions too. Do you see any possible issue?